### DISCUSSION TITLE
Installing slither in an isolated environment

### DISCUSSION THREAD

**Discussion by PatrickAlphaC** [crytic/slither#1728] [Q&A]
### Describe the desired feature

Slither has very specific dependencies in it's current form, and I've found much greater success working with it in it's own isolated environment. One way to do it is to install slither with [pipx](https://pypa.github.io/pipx/) instead of `pip` 

Please consider adding this to the docs.

Step 1. [Install pipx](https://pypa.github.io/pipx/)

Step 2. Install slither

```
pipx install slither-analyzer
```

Step 3. Run slither

```
slither .
```

**Comment by 0xalpharush** [comment]
There are many different tools to isolate python installations. We've documented using slither with `pyenv` and `virtualenv` here https://github.com/crytic/slither/wiki/Developer-installation#developer-installation-using-virtualenvs

**Reply by PatrickAlphaC** [reply]
Yep! There are a few, I'm just a big fan of `pipx`, as I think it's less cognitive load and easier for people to use& understand.

**Reply by ghost** [reply]
Hi, when I'm trying to install slither using command `pipx install slither-analyzer`, i got some errors as shown below:

```bash
$ pipx install slither-analyzer
Fatal error from pip prevented installation. Full pip output in file:
    C:\Users\User\pipx\logs\cmd_2025-03-06_11.09.19_pip_errors.log

pip failed to build package:
    ckzg

Some possibly relevant errors from pip install:
    error: subprocess-exited-with-error
    LINK : fatal error LNK1181: cannot open input file 'blst.lib'
    error: command 'C:\\Program Files (x86)\\Microsoft Visual Studio\\2022\\BuildTools\\VC\\Tools\\MSVC\\14.43.34808\\bin\\HostX86\\x64\\link.exe' failed with exit code 1181
    ERROR: Failed to build installable wheels for some pyproject.toml based projects (ckzg)

Error installing slither-analyzer.
```

any help would be greatly appreciated
